src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/ViewResultExecutor.cs

```diff
@@ -68,7 +68,12 @@ public virtual ViewEngineResult FindView(ActionContext actionContext, ViewResult
             var viewEngine = viewResult.ViewEngine ?? ViewEngine;
             var viewName = viewResult.ViewName ?? actionContext.ActionDescriptor.Name;
 
-            var result = viewEngine.FindView(actionContext, viewName);
+            var result = viewEngine.GetView(executingFilePath: null, viewPath: viewName, isPartial: false);
```

Comment:
Yes, moving the check(s) one layer up was exactly the point of @Eilon's request. We can now easily add for example `LayoutPath` and `LayoutName` properties that expose this separation to users.